app/src/main/kotlin/com/wire/android/ui/WireActivityViewModel.kt

```diff
@@ -434,10 +425,11 @@ class WireActivityViewModel @Inject constructor(
 
     fun shouldLogIn(): Boolean = !hasValidCurrentSession()
 
-    fun blockedByE2EI(): Boolean {
-        return observeE2EIState.value == true
+    private fun blockedByE2EI(): Boolean = runBlocking {
```

Comment:
question: do we need it to runBlocking can't we have it return a `Deferred` and await the result when needed ?

Reply:
We reviewed online and it's resolved.